app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -503,8 +513,18 @@ def vm_instance_defaults
         :volumes    => [new_volume],
         :scsi_controller => { :type => scsi_controller_default_type },
         :datacenter => datacenter,
+        :firmware => 'automatic',
         :boot_order => ['network', 'disk']
       )
     end
+
+    def firmware_mapping(firmware)
+      case firmware
+      when /UEFI/
```

Comment:
Feel free to create a helper in the pxe_loader helper class `PxeLoaderSupport` (and a unit test) for that. That's more appropriate place than here.